source/agora/node/Runner.d

```diff
@@ -129,11 +131,20 @@ public Listeners runNode (Config config)
 
     setTimer(0.seconds, &result.node.start, Periodic.No);  // asynchronous
 
-    auto tls_ctx = getTLSContext();
+    auto tls_ctx = getTLSContext(log);
+    if (tls_ctx)
+        log.info("Node is setup to enable https");
+    else
+        log.info("Node is NOT setup to enable https");
     if (result.admin !is null)
     {
+        bool checkPassword(string user, string password) @safe
+        {
+            return user == config.admin.username && password == config.admin.password;
+        }
         log.info("Admin interface listening will be on {}:{}", config.admin.address, config.admin.port);
         auto adminrouter = new URLRouter();
+        adminrouter.any("*", performBasicAuth("Site Realm", toDelegate(&checkPassword)));
```

Comment:
Need a more friendly title than "Site Realm"

Reply:
Don't think you need `toDelegate` here

Reply:
Updated